### PR TITLE
Put Host into a header for Forward Auth

### PR DESCRIPTION
**Pull request by ekozan**
Hi all,

I just added possibility of forward the origin host to the as

Config example 

```toml
[entryPoints.http.auth.forward]
address = "http://authtest/"
hostheader = "x-origin-host"
```

It's follow the design  
> Additional Headers
> The AS might be interested in some of the original request informations such as:
> Original Host - The target host address, eg api.example.com

> All this information should be optional. The user must be allowed to define each header name. 

Related to #2105

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -341,8 +341,14 @@ type Digest struct {
 
 // Forward authentication
 type Forward struct {
-	Address string     `description:"Authentication server address"`
-	TLS     *ClientTLS `description:"Enable TLS support"`
+	Address         string           `description:"Authentication server address"`
+	TLS             *ClientTLS       `description:"Enable TLS support"`
+	ForwardHostname *ForwardHostname `description:"Enable forward hostname request "`
```
Comment (ldez) [inline types/types.go]: You simply like that:
```go
// Forward authentication
type Forward struct {
	Address         string           `description:"Authentication server address"`
	TLS             *ClientTLS       `description:"Enable TLS support"`
	HostHeader      string           `description:"Header name to put the forwarded host"`
}
```

Comment (ekozan) [reply types/types.go]: done.

**Comment by ldez** [comment]
@ekozan After discussion with the team, we believe that this PR should go in version 1.4.

Can you change the root branch of PR and rebase on branch 1.4 ?
Ask me if yo need help.

**Comment by ldez** [comment]
After the rebase you will need to add some documentation https://github.com/containous/traefik/blob/v1.4/docs/configuration/entrypoints.md#forward-authentication

**Comment by ekozan** [comment]
done. :)

**Comment by ldez** [comment]
could you add some documentation https://github.com/containous/traefik/blob/v1.4/docs/configuration/entrypoints.md#forward-authentication

**Comment by ekozan** [comment]
Done, I'm not really good for write doc :P
let me know if it's not good enough

**Comment by emilevauge** [comment]
Hey @ekozan,
Thanks a lot for this contribution :heart: 
After digging into this, it appears that a bigger fix is necessary and we decided [to do it ourselves](https://github.com/containous/traefik/pull/2132/).
I hope you don't mind if we close this one in favor of the new one :) OK for you ?
Thanks again !

**Comment by ekozan** [comment]
:) okay

**Comment by abouzian** [comment]
Hello guys, anyone can help with using [entryPoints.http.auth.forward] in  [entryPoints.https] ?
How to forward authentication when using HTTPS ??
Thx.

**Comment by magnock** [comment]
Hello guys, anyone can help with using [entryPoints.http.auth.forward] in [entryPoints.https] ?
How to forward authentication when using HTTPS ??
Thx.

**Pull request closed without merging**